src/parsers/minecraft/coord/coord_base.ts

```diff
@@ -0,0 +1,129 @@
+import { CommandErrorBuilder } from "../../../brigadier_components/errors";
+import { StringReader } from "../../../brigadier_components/string_reader";
+import { ReturnHelper } from "../../../misc_functions";
+import { Parser, ReturnedInfo } from "../../../types";
+
+const MIXED = new CommandErrorBuilder(
+    "argument.pos.mixed",
+    "Cannot mix world & local coordinates (everything must either use ^ or not)"
+);
+
+const INCOMPLETE = new CommandErrorBuilder(
+    "argument.pos.incomplete",
+    "Incomplete position argument. Only %s coords are present, when %s should be"
+);
+
+const NO_LOCAL = new CommandErrorBuilder(
+    "argument.pos.nolocal",
+    "Local coords are not allowed"
+);
+
+export class CoordBaseParser implements Parser {
```

Comment:
Doesn't really need to be called `CoordBaseParser`, `CoordParser` would be fine.